lib/services/local/events.go

```diff
@@ -1961,7 +1963,30 @@ type userNotificationParser struct {
 func (p *userNotificationParser) parse(event backend.Event) (types.Resource, error) {
 	switch event.Type {
 	case types.OpDelete:
-		return resourceHeader(event, types.KindNotification, types.V1, 0)
+		// Remove the first separator so none of the separated parts will be
+		// empty strings
+		key := string(event.Item.Key)
+		if len(key) > 0 && key[0] == backend.Separator {
+			key = key[1:]
+		}
+		parts := strings.Split(key, string(backend.Separator))
+		if len(parts) != 4 {
+			return nil, trace.BadParameter("malformed key for %s event: %s", types.KindNotification, event.Item.Key)
+		}
+
+		notification := &notificationsv1.Notification{
+			Kind:    types.KindNotification,
+			Version: types.V1,
+			Spec: &notificationsv1.NotificationSpec{
+				Username: parts[2],
+				Id:       parts[3],
+			},
+			Metadata: &headerv1.Metadata{
+				Name: parts[3],
+			},
+		}
+
+		return types.Resource153ToLegacy(notification), nil
```

Comment:
@fspmarshall why have we only started using a full resource skeleton for OpDelete in the last couple of resource types? Is there something that will fail if we return the resource rather than a ResourceHeader of some kind?

Reply:
It may not be the exact answer to your question @espadolini , but I remember `OpDelete` giving some issue with the unified resource cache. When we put an item, we were able to generate keys anyway we want (with stuff like `GetHostName` or `GetApp().GetName()`. Without all the information sent during `OpDelete` I had troubles generating the keys to be deleted. With a couple other things moving to the sortcache api, perhaps this is because of that?

Reply:
I imagine it was for consistency's sake.  I don't know of any particular reason `OpDelete` _needs_ to be `ResourceHeader`, other than maybe to avoid people accidentally making the assumption that the _entire_ resource is available when only a subset is.